src/Nethermind/Nethermind.Evm.Test/Eip4844Tests.cs

```diff
@@ -0,0 +1,63 @@
+// SPDX-FileCopyrightText: 2022 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using FluentAssertions;
+using Nethermind.Specs;
+using NUnit.Framework;
+using System;
+using Nethermind.Int256;
+using System.Linq;
+
+namespace Nethermind.Evm.Test
+{
+    [TestFixture]
+    public class Eip4844Tests : VirtualMachineTestsBase
+    {
+        protected override long BlockNumber => MainnetSpecProvider.GrayGlacierBlockNumber;
+        protected override ulong Timestamp => MainnetSpecProvider.ShardingForkBlockTimestamp;
+
+
+        [TestCase(0, 0, Description = "Should return 0 when no hashes")]
+        [TestCase(1, 1, Description = "Should return 0 when out of range")]
+        [TestCase(2, 1, Description = "Should return 0 when way out of range")]
+        [TestCase(0, 1, Description = "Should return hash, when exists")]
+        [TestCase(1, 3, Description = "Should return hash, when exists")]
+        public void Test_datahash_index_in_range(int index, int datahashesCount)
+        {
+            byte[][] hashes = new byte[datahashesCount][];
+            for (int i = 0; i < datahashesCount; i++)
+            {
+                hashes[i] = new byte[32];
+                for (int n = 0; n < datahashesCount; n++)
+                {
+                    hashes[i][n] = (byte)((i * 3 + 10 * 7) % 256);
+                }
+            }
+            byte[] expectedOutput = datahashesCount > index ? hashes[index] : new byte[32];
+
+            // Cost of transaction call + PUSH1 x4 + MSTORE
+            const long GasCostOfCallingWrapper = GasCostOf.Transaction + GasCostOf.VeryLow * 6;
```

Comment:
Why is it `GasCostOf.VeryLow * 6` when 4x `PUSH1` + 1x `MSTORE` sum to 5x?

Reply:
MSTORE requires Low=3 entry cost + variable cost depending on position in memory and length, which is just Memory=3, because it's at the very beginning: https://github.com/NethermindEth/nethermind/blob/e262197731eddbeb0adec62a35c531ededece59e/src/Nethermind/Nethermind.Evm/EvmPooledMemory.cs#L187
I added this detail to the comment

Reply:
Thanks for explanation, everything clear now! :)